packages/vite-plugin-svelte/src/utils/optimizer-plugins.js

```diff
@@ -49,11 +68,63 @@ export function esbuildSveltePlugin(options) {
 	};
 }
 
+/**
+ * @param {import('../types/options.d.ts').ResolvedOptions} options
+ * @returns {RolldownPlugin}
+ */
+export function rolldownOptimizeSveltePlugin(options) {
+	/** @type {import('../types/vite-plugin-svelte-stats.d.ts').StatCollection | undefined} */
+	let statsCollection;
+	/** @type {RolldownPlugin} */
+	const plugin = {
+		name: 'vite-plugin-svelte:rolldown-optimize-svelte',
+		// @ts-expect-error not typed in rolldown yet
+		options(opts) {
+			if (
+				opts.plugins?.some((/** @type {{ name: string; }} */ p) =>
+					p.name.startsWith('vite:dep-scan')
+				)
+			) {
+				delete plugin.transform;
+				delete plugin.buildStart;
+				delete plugin.buildEnd;
+			}
+		},
+		transform: {
+			filter: {
+				// TODO: remove excludes once above options hook works
+				id: { include: [/^[^?#]+\.svelte(?:[?#]|$)/], exclude: [/^virtual-module:/] },
+				code: { exclude: [/(?:import|export)[^"\n]+"virtual-module:/] }
+			},
+			/**
+			 * @param {string} code
+			 * @param {string} filename
+			 */
+			async handler(code, filename) {
+				try {
+					return await compileSvelte(options, { filename, code }, statsCollection);
```

Comment:
instead of esbuild plugin that returns sourcemaps inline with a comment, this returns code,map.